index.js

```diff
@@ -136,49 +178,75 @@ function nowServing(line) {
 }
 
 const line = []
+const extraLine = []
 
 // ***** Scope & Closures - Question 1 *****
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
+
+function takeATicketNumber(line){
+  let nextNum = extraLine.length+1
+  line.push(nextNum)
+  extraLine.push(nextNum)
+  return `Welcome, You are ticket number ${nextNum}`
+}
+
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 1.`
 
-// console.log(nowServing(line))
+console.log(nowServing(line))
 // // => `Currently serving 2.`
 
-// console.log(takeATicketNumber(line))
+console.log(takeATicketNumber(line))
 // // => `Welcome. You are ticket number 3`
 
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 // ***** Scope & Closures - Question 2 *****
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
+
+function ticketNumberGeneratorFunc(){
+  const extraLine = []
+  const line = []
+
+  function takeATicketNumber2(line){
+    let nextNum = extraLine.length+1
+    line.push(nextNum)
+    extraLine.push(nextNum)
+    return `Welcome, You are ticket number ${nextNum}`
+  }
+
+  return takeATicketNumber2
+}
+
+const newLine = [] // goals is NOT to use a global variable to store data
+const takeATicketNumberFunc = ticketNumberGeneratorFunc()
```

Comment:
Consider the above comment as well for this one in relation to `extraLine`. You can still use the global variable `newLine` on line 233 (the directions were hinting at a solution for this problem while hinting at the solution for the previous problem).